src/Uno.UI/Microsoft/UI/Xaml/Controls/NumberBox/NumberBox.Properties.cs

```diff
@@ -40,16 +40,22 @@ public double Maximum
 			set => SetValue(MaximumProperty, value);
 		}
 
-		public static DependencyProperty MaximumProperty { get ; } =
+		public static DependencyProperty MaximumProperty { get; } =
 			DependencyProperty.Register(nameof(Maximum), typeof(double), typeof(NumberBox), new FrameworkPropertyMetadata(double.MaxValue, (s, e) => (s as NumberBox)?.OnMaximumPropertyChanged(e)));
 
 		public double Value
 		{
 			get => (double)GetValue(ValueProperty);
-			set => SetValue(ValueProperty, value);
+			set
+			{
+				if (!Value.IsNaN() || !value.IsNaN())
```

Comment:
How about check IsNegative ?

Reply:
@lindexi This is ported from microsoft-ui-xaml repository.